### PR TITLE
Fetch Enhancements

### PR DESCRIPTION
**Pull request by trevorgerhardt**
Added
- Multiple fetching
- Automatic retrying

Changed
- Switched `next` callback to `next(error, response)`

### CODE REVIEW

**Comment by trevorgerhardt** [comment]
I could make this a BREAKING CHANGE and call it 1.0....I'm pretty satisfied with where it's at.

File: src/fetch.js [hunk 1]
```diff
@@ -44,10 +51,44 @@ function runFetch ({
   })
     .then(checkStatus)
     .then(createResponse)
-    .then((response) => [decrementFetches(), next(response)])
-    .catch((error) =>
-      createErrorResponse(error)
-        .then((response) => [decrementFetches(), fetchError(response), next(response, true)]))
+    .then(async (response) =>
+      (retry && await retry(response))
```
Comment (evansiroky) [inline src/fetch.js]: How does this work?

**Comment by evansiroky** [comment]
Can you expand the documentation on fetch in the README?  Also seems kind of unclear how the automatic retries work.

File: __tests__/lib/fetch.js [hunk 2]
```diff
@@ -1,36 +1,71 @@
...
+      next: done
     })
-    await store.dispatch(action)
-    expect(action.payload.next).toBeCalled()
+    store.dispatch(action)
   })
 
-  it('should automatically parse json responses', async () => {
+  it('should automatically parse json responses', (done) => {
+    nock('http://google.com')
```
Comment (evansiroky) [inline __tests__/lib/fetch.js]: This nock will not affect the following request

File: src/fetch.js (hunk 1 above)
Comment (trevorgerhardt) [inline src/fetch.js]: `retry` is a function that receives that response and can return either a Boolean or a Promise that evaluates to a Boolean.

File: __tests__/lib/fetch.js (hunk 2 above)
Comment (trevorgerhardt) [inline __tests__/lib/fetch.js]: Accidentally didn't update the URL, good catch.

File: README.md [hunk 3]
```diff
@@ -38,7 +38,56 @@ This will create a redux store with the `fetch`, `history`, `logger`, `multi`, a
...
+### `fetchMultiple({fetches, next})`
+
+Allows you to dispatch a single action that will call next with all of the responses.
+
+```js
+const {fetchMultiple} = require('@conveyal/woonerf/fetch')
+
+store.dispatch(fetchMultiple({
+  fetches: [
+    url: 'http://conveyal.com',
```
Comment (evansiroky) [inline README.md]: This is not valid json

Comment (trevorgerhardt) [inline README.md]: 👌 fixed

**Pull request merged**